shared/utils/typedefs-string.hpp

```diff
@@ -102,19 +117,110 @@ struct StringW {
     constexpr StringW(std::nullptr_t npt) noexcept : inst(npt) {}
     constexpr StringW() noexcept : inst(nullptr) {}
 
-    operator std::string();
+    constexpr void const* convert() const noexcept {
+        return inst;
+    }
     constexpr void* convert() noexcept {
         return inst;
     }
+    constexpr operator Il2CppString const*() const noexcept {
+        return inst;
+    }
     constexpr operator Il2CppString*() noexcept {
         return inst;
     }
+    constexpr Il2CppString const* operator->() const noexcept {
+        return inst;
+    }
     constexpr Il2CppString* operator->() noexcept {
         return inst;
     }
-    operator std::u16string();
-    operator std::wstring();
+    constexpr operator bool() const noexcept {
+        return inst != nullptr; 
+    }
+
+    // the argument here is unused but we don't care. the comparison against nullptr more clearly shows what this operator is for imo
```

Comment:
try using `[[maybe_unused]]` or just not giving the parameter a name, though you should actually compare it to the parameter to be clear.